docs/best-practices.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: 'Best Practices'
+---
+
+Emotion is an extremely flexible library, but this can make it intimidating, especially for new users. This guide contains several recommendations for how to use Emotion in your application. Keep in mind, these are only recommendations, not requirements!
+
+## Recommendations
+
+### Use TypeScript and object styles
+
+You don't get Intellisense or type checking when using CSS strings, e.g. `` css`color: blue` ``. You can improve the developer experience and prevent style bugs by using TypeScript and object styles, which enable Intellisense and _some_ static type checking:
+
+```tsx
+const myCss = css({
+  color: 'blue',
+  grid: 1 // Error: Type 'number' is not assignable to type 'Grid | Grid[] | undefined'
+})
+```
+
+### Share styles via component reuse
+
+Suppose you are developing an application that needs to display error messages in many different components. All of the error messages should be red and bold. One way to accomplish this is by exporting the CSS from a shared file and importing the CSS in many places:
+
+```tsx
+export const errorCss = css({
+  color: 'red',
+  fontWeight: 'bold'
+})
+
+// Then, in one of your components:
+import { errorCss } from '...'
+
+return <p css={errorCss}>Failed to fizzle the frozzle.</p>
+```
+
+While there's absolutely nothing functionally wrong with this, we recommend sharing styles with component reuse in most cases. For example,
+
+```tsx
+export function ErrorMessage({ className, children }) {
+  return (
+    <p css={{ color: 'red', fontWeight: 'bold' }} className={className}>
+      {children}
+    </p>
+  )
+}
+
+// Then, in one of your components
+import { ErrorMessage } from '...'
+
+return <ErrorMessage>Failed to fizzle the frozzle.</ErrorMessage>
+```
+
+One advantage of this approach is that you can add additional props and functionality to the `ErrorMessage` component with limited refactoring.
+
+### Colocate styles with components
+
+With normal CSS, the styles for a component are defined in a separate file. This makes maintenance more difficult because it's harder to tell which components use a given piece of CSS, and you can easily forget to update the relevant CSS after modifying a component. One of the main benefits of Emotion is that you can [colocate](https://kentcdodds.com/blog/colocation) your styles with your components. All this means is that the CSS for a component should be in the same file as the component itself. This point is closely related to sharing styles via component reuse.
+
+### Use the `style` prop for dynamic styles
```

Comment:
this is a very important one, thank you for adding this! perhaps we should mention using CSS vars defined in `style` here too? I'm not sure if css vars are that much needed in this context but i guess keeping all style declarations in a single place and just "deferring" their actual values for the `style` pop might make sense for readability etc

Reply:
Added a section that explains this 👍